File: src/org/jetbrains/haskell/fileType/HaskellFileType.java
package org.jetbrains.haskell.fileType;

import com.intellij.openapi.fileTypes.LanguageFileType;
import com.intellij.openapi.util.IconLoader;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.haskell.HaskellLanguage;
import org.jetbrains.haskell.icons.HaskellIcons;

import javax.swing.*;

public class HaskellFileType extends LanguageFileType {
    public static final HaskellFileType INSTANCE = new HaskellFileType();
    @NonNls
    public static final String DEFAULT_EXTENSION = "hs";

    private Icon myIcon;

    private HaskellFileType() {
        super(HaskellLanguage.INSTANCE);
        myIcon = HaskellIcons.DEFAULT;
    }

    @NotNull
    public String getName() {
        return "Haskell file";
    }

    @NotNull
    public String getDescription() {
        return "Haskell file";
    }

    @NotNull
    public String getDefaultExtension() {
        return DEFAULT_EXTENSION;
    }

    public Icon getIcon() {
        return myIcon;
    }

    public String getCharset(@NotNull VirtualFile file, final byte[] content) {
        return "UTF-8";
    }
}

File: plugin/src/org/jetbrains/haskell/fileType/HaskellFileType.kt
package org.jetbrains.haskell.fileType

import com.intellij.openapi.fileTypes.LanguageFileType
import com.intellij.openapi.vfs.VirtualFile
import org.jetbrains.annotations.NonNls
import org.jetbrains.haskell.HaskellLanguage
import org.jetbrains.haskell.icons.HaskellIcons
import javax.swing.*

class HaskellFileType : LanguageFileType(HaskellLanguage.INSTANCE) {

    private var myIcon: Icon = HaskellIcons.DEFAULT

    override fun getName(): String =
            "Haskell file"

    override fun getDescription(): String =
            "Haskell file"

    override fun getDefaultExtension(): String =
            DEFAULT_EXTENSION

    override fun getIcon(): Icon =
            myIcon

    override fun getCharset(file: VirtualFile, content: ByteArray): String =
            "UTF-8"

    companion object {
        @JvmField val INSTANCE: HaskellFileType = HaskellFileType()
        val DEFAULT_EXTENSION: String = "hs"
    }
}
